crates/matrix-sdk/src/event_cache/mod.rs

```diff
@@ -396,6 +404,49 @@ impl EventCacheInner {
                 continue;
             };
 
+            // Handle and cache events and relations
+            let mut relations = self.relations.write().await;
```

Comment:
Unfortunately, this is not the right place to add this; I suppose this should be in the same place where I inserted events into `Self::events`. Follow the breadcrumbs :)

Reply:
I think I'll have to do that in a separate method then? I don't think I can have 2 `RwLockWriteGuards` at the same time, one for `all_events` and another one for `relations`, so I don't think I'll be able to add this on `append_events_locked_impl`.

Reply:
Oh, I just saw https://github.com/matrix-org/matrix-rust-sdk/pull/3840#discussion_r1716841925.